.gitignore

```diff
@@ -6,3 +6,7 @@
 /docs/generated
 /docs/doxyoutput
 /docs/api
+tags
+**/generated/
+**/results/
```

Comment:
Can these be fixed to only be ignored under testing dirs?

Reply:
`results` appears to be created on the top level, so I left that in here. Moved the rest to `testing/.gitignore`.